internal/grpc/grpc.go

```diff
@@ -77,7 +91,65 @@ func GetDialOptions(agentConfig *config.Config) []grpc.DialOption {
 	return opts
 }
 
-// nolint: ireturn
+// Have to create our own UnaryClientInterceptor function since protovalidate only provides a UnaryServerInterceptor
+// https://pkg.go.dev/github.com/grpc-ecosystem/go-grpc-middleware/v2@v2.1.0/interceptors/protovalidate
+func ProtoValidatorUnaryClientInterceptor() (grpc.UnaryClientInterceptor, error) {
+	validator, err := protovalidate.New()
+	if err != nil {
+		return nil, err
+	}
+
+	return func(
+		ctx context.Context,
+		method string,
+		req, reply interface{},
+		cc *grpc.ClientConn,
+		invoker grpc.UnaryInvoker,
+		opts ...grpc.CallOption,
+	) error {
+		slog.Debug("Validation interceptor request", "req", req)
+
+		requestValidationErr := validateMessage(validator, req)
+		if requestValidationErr != nil {
+			return status.Errorf(
+				codes.InvalidArgument,
+				fmt.Errorf("invalid request message: %w", requestValidationErr).Error(),
```

Comment:
nit: "Errorf returns Error(c, fmt.Sprintf(format, a...))."
```suggestion
				"invalid request message: %s", requestValidationErr.Error(),
```